### PR TITLE
feat: Stale return value instead of default

### PR DESCRIPTION
**Pull request by vahidlazio**

### CODE REVIEW

**Review by nicklasl — DISMISSED** [review]

**Comment by nicklasl** [comment]
Seems we had no tests for this? Can you add some?

**Comment by vahidlazio** [comment]
@nicklasl added test.

File: Provider/src/test/java/com/spotify/confidence/ConfidenceFeatureProviderTests.kt [hunk 1]
```diff
@@ -543,6 +543,50 @@ internal class ConfidenceFeatureProviderTests {
...
+        advanceUntilIdle()
+
+        verify(flagResolverClient, times(1)).resolve(any(), eq(context))
+
+        val evalString = confidenceFeatureProvider.getStringEvaluation(
+            "test-kotlin-flag-1.mystring",
+            "default",
+            evaluationContext
+        )
+        assertEquals(evalString.reason, Reason.TARGETING_MATCH.name)
```
Comment (nicklasl) [inline Provider/src/test/java/com/spotify/confidence/ConfidenceFeatureProviderTests.kt:579]:
```suggestion
        assertEquals(evalString.reason, Reason.TARGETING_MATCH.name)
        assertEquals(evalString.value, "red")
```

**Review by nicklasl — APPROVED** [review]

File: Provider/src/test/java/com/spotify/confidence/ConfidenceFeatureProviderTests.kt [hunk 2]
```diff
@@ -543,6 +543,52 @@ internal class ConfidenceFeatureProviderTests {
...
+        assertEquals(evalString.value, "red")
+        
+        mockConfidence.putContext("hello", ConfidenceValue.String("new context"))
```
Comment (nicklasl) [inline Provider/src/test/java/com/spotify/confidence/ConfidenceFeatureProviderTests.kt:582]:
```suggestion
        assertEquals(evalString.reason, Reason.TARGETING_MATCH.name)
        assertEquals(evalString.value, "red")

        mockConfidence.putContext("hello", ConfidenceValue.String("new context"))
```

**Review by nicklasl — APPROVED** [review]

**Pull request merged**